openforcefield/tests/test_molecule.py

```diff
@@ -1053,6 +1053,173 @@ def test_wrong_index_mapping(self):
+            # test the max molecules option
+            mol = Molecule.from_smiles('c1[nH]c2c(=O)[nH]c(nc2n1)N', toolkit_registry=toolkit, allow_undefined_stereo=True)
+
+            tauts_no = 5
+            tautomers = mol.enumerate_tautomers(max_states=tauts_no, toolkit_registry=toolkit)
+            assert len(tautomers) <= tauts_no
+            assert mol not in tautomers
+
+    @pytest.mark.parametrize('toolkit_class', [RDKitToolkitWrapper, OpenEyeToolkitWrapper])
+    def test_enumerating_no_tautomers(self, toolkit_class):
+        """Test that the toolkits return an empty list if there are no tautomers to enumerate."""
+
+        if toolkit_class.is_available():
+            toolkit = toolkit_class()
+            mol = Molecule.from_smiles('CC', toolkit_registry=toolkit)
+
+            tautomers = mol.enumerate_tautomers(toolkit_registry=toolkit)
+            assert tautomers == []
+
+        else:
+            pytest.skip('Required toolkit is unavailable')
+
+    @requires_openeye
+    def test_enumerating_no_formalcharges(self):
+        """Make sure no protomers are returned."""
+
+        mol = Molecule.from_smiles('CC')
+
+        assert mol.enumerate_formalcharges() == []
+
+    @requires_openeye
+    def test_enumerating_formalcharges(self):
+        """Test enumerating the formal charges."""
+
+        mol = Molecule.from_smiles('Oc2ccc(c1ccncc1)cc2')
+
+        # there should be three protomers for this molecule so restrict the output
+        protomers = mol.enumerate_formalcharges(max_states=2)
+
+        assert mol not in protomers
+        assert len(protomers) == 2
+
+        # now make sure we can generate them all
+        protomers = mol.enumerate_formalcharges(max_states=10)
+
+        assert mol not in protomers
+        assert len(protomers) == 3
+
+        # make sure the molecules are different
+        for protomer in protomers:
+            assert mol != protomer
+
+    @pytest.mark.parametrize('toolkit_class', [OpenEyeToolkitWrapper, RDKitToolkitWrapper])
+    def test_enumerating_stereobonds(self, toolkit_class):
+        """Test the backend toolkits in enumerating the stereo bonds in a molecule."""
+
+        if toolkit_class.is_available():
+            toolkit = toolkit_class()
+            mol = Molecule.from_smiles('ClC=CCl', allow_undefined_stereo=True, toolkit_registry=toolkit)
+
+            # use the default options
+            isomers = mol.enumerate_stereoisomers()
+            assert len(isomers) == 2
+
+            assert mol not in isomers
+            # make sure the input molecule is only different by bond stereo
+            for ismol in isomers:
+                assert Molecule.are_isomorphic(mol, ismol, return_atom_map=False, bond_stereochemistry_matching=False)[0] is True
+                assert mol.is_isomorphic_with(ismol) is False
+
+            # make sure the isomers are different
+            assert Molecule.are_isomorphic(isomers[0], isomers[1])[0] is False
```

Comment:
Just a little cleaner
```suggestion
            assert isomers[0].is_isomorphic_with(isomers[1]) is False
```